compiler/noirc_evaluator/src/ssa/ir/instruction.rs

```diff
@@ -242,20 +242,11 @@ impl Instruction {
         matches!(self.result_type(), InstructionResultType::Unknown)
     }
 
-    /// Pure `Instructions` are instructions which have no side-effects and results are a function of the inputs only,
-    /// i.e. there are no interactions with memory.
-    ///
-    /// Pure instructions can be replaced with the results of another pure instruction with the same inputs.
-    pub(crate) fn is_pure(&self, dfg: &DataFlowGraph) -> bool {
+    /// Indicates if the instruction can be safely replaced with the results of another instruction with the same inputs.
+    pub(crate) fn can_be_replaced(&self, dfg: &DataFlowGraph) -> bool {
```

Comment:
```suggestion
    pub(crate) fn can_be_deduplicated(&self, dfg: &DataFlowGraph) -> bool {
```
`can_be_replaced` made me think "replaced by what?"

Reply:
Done